demo/src/test/java/com/mongodb/jdbc/utils/TestUtils.java

```diff
@@ -0,0 +1,439 @@
+                + m_dbMeta.getJDBCMajorVersion());
+        System.out.println("get Identifier Quote String: "
+                + m_dbMeta.getTimeDateFunctions());
+        System.out.println("get Identifier Quote String: "
+                + m_dbMeta.getStringFunctions());
+        System.out.println("get Identifier Quote String: "
+                + m_dbMeta.getNumericFunctions());
+    }
+
+    @Test
+    void testGetUrl() throws Exception
+    {
+        System.out.println(m_dbMeta.getURL());
+    }
+
+    @Test
+    void testSetCatalog() throws SQLException
+    {
+        m_conn.setCatalog("test_set_catalog");
+        System.out.println(m_conn.getCatalog());
+    }
+
+    @Test
+    // Note : Cancel is not supported yet
+    void testCancel() throws SQLException
+    {
+        // The query must be a long running query for cancel to be effective
+        String query = "select * from class";
+        System.out
+                .println("-----------------------------------------------------------------------");
+
+        class QueryExecutor implements Runnable
+        {
+            private boolean m_isExecuting;
+            private final String m_query;
+            private ResultSet m_result;
+            private final Statement m_statement;
+
+            public QueryExecutor(Statement statement, String query, ResultSet result)
+            {
+                m_statement = statement;
+                m_query = query;
+                m_result = result;
+                m_isExecuting = false;
+            }
+
+            public boolean getIsExecuting()
+            {
+                return m_isExecuting;
+            }
+
+            public ResultSet getResultSet()
+            {
+                return m_result;
+            }
+
+            @Override
+            public void run()
+            {
+                try
+                {
+                    System.out.println("Executing '" + m_query + "'");
+                    m_isExecuting = true;
+                    m_result = m_statement.executeQuery(m_query);
+                    m_isExecuting = false;
+                }
+                catch (SQLException e)
+                {
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        Statement statement = m_conn.createStatement();
+        ResultSet result = null;
+
+        QueryExecutor queryExecutor = new QueryExecutor(statement, query, result);
+        Thread execThread = new Thread(queryExecutor);
+        execThread.start();
+
+        while (!queryExecutor.getIsExecuting())
+        {
+            ;
+        }
+        if (execThread.isAlive())
+        {
+            try
+            {
+                // Still need to wait because the m_isExecuting flag is set
+                // before
+                // statement.executeQuery is called. Need to wait until an
+                // executor
+                // object is created.
```

Comment:
The newline placement looks a little odd here
```suggestion
                // Still need to wait because the m_isExecuting flag is set
                // before statement.executeQuery is called. Need to wait until
                // an executor object is created.
```